### ISSUE TITLE
Error "cannot update cluster data: Unable to complete atomic operation, key modified"

### ISSUE THREAD

**Issue by nh2** [sorintlab/stolon#487]
Stolon `v0.10.0`, I'm getting this error:

```
cannot update cluster data: Unable to complete atomic operation, key modified
```

According to this code, getting this error message should be impossible:

https://github.com/sorintlab/stolon/blob/d07f290bcfabc0d5292c9b072509880e7d99f246/cmd/stolonctl/cmd/update.go#L131-L139

because `if err == store.ErrKeyModified { continue }`, so the `die()` shouldn't be reachable.

The problem is that there are 2 definitions of `ErrKeyModified`:

* https://github.com/sorintlab/stolon/blob/d07f290bcfabc0d5292c9b072509880e7d99f246/pkg/store/store.go#L28
* https://github.com/sorintlab/stolon/blob/d07f290bcfabc0d5292c9b072509880e7d99f246/vendor/github.com/docker/libkv/store/store.go#L33

Stolon compares to the first one, but the error raised here is the second one from `vendor/github.com/docker/libkv`.

---

## Workaround

This patch is a workaround, because the two error definitions have equal string messages inside:

```diff
- if err == store.ErrKeyModified {
+ if err.Error() == store.ErrKeyModified.Error() {
```

**Comment by nh2** [comment]
This printout further confirms it:

```go
                        fmt.Println("equals error:", err == store.ErrKeyModified)
                        fmt.Println("equals2 error:", err.Error() == store.ErrKeyModified.Error())
```

prints:

```
equals error: false
equals2 error: true
```

**Comment by sgotti** [comment]
@nh2 Good catch. If you look at https://github.com/sorintlab/stolon/blob/d07f290bcfabc0d5292c9b072509880e7d99f246/pkg/store/libkv.go#L31 every call to libkv converts the error to `store.ErrKeyNotFound but we are not converting `ErrKeyModified`. So the real fix will be to convert also it inside that funciont. Care to open a PR?

**Comment by nh2** [comment]
@sgotti I could put the extra cases in, but is there a way to do exhaustive case matching in Go so that we'd get a warning or compile error when some are missing?

Also, why does `store.go` have its own equivalents with same name and message? Would it be sensible to use libkv's ones, always?

**Comment by sgotti** [comment]
PR #496

**Comment by sgotti** [comment]
Fixed in #496.
